exercise/cache/hierarchy.py

```diff
@@ -160,8 +160,18 @@ def _go_to_last(indices: List[int]):
 
     if skip_first:
         # Move the starting index to the previous element
-        if tree_indices[-1] == 0:
-            tree_indices = tree_indices[-1]
+        lastidx = len(tree_indices) - 1
+        while tree_indices[lastidx] == 0:
+            if lastidx < 1:
+                tree_indices = 0
+                break
+            if tree_indices[lastidx - 1] > 0:
+                # if we are at the first chapter at the current level, move to previous item at
+                # next highest level (or move further up the tree, if not possible)
+                tree_indices[lastidx - 1] -= 1
+                tree_indices.pop()
+                break
+            lastidx -= 1
```

Comment:
I think there is a bug somewhere here. I was testing with the Aplus manual. When I open chapter 2.1 First steps, the previous link points to the index of module 1 Overview, but it should point to the module 2 index. Similarly in the chapter 3.1, the previous link points to the module 2 instead of 3.

Reply:
This should be fixed now in the new version.